### ISSUE TITLE
command 'slides.toggle' not found

### ISSUE THREAD

**Issue by marc-fez** [nicoespeon/vscode-slides#45]
Hello,
I've seen [this issue](https://github.com/nicoespeon/vscode-slides/issues/31) and have the same problem. I see that the original defect was solved by removing extensions, woulda been cool if they had mentioned which extensions, but whatever.

I've installed the Slides extension, the recommended theme and font. Ctrl-Alt-P displays the error.
Running Toggle Slides Mode from the command palette also shows the same error

Version: 1.63.2
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:39:46.686Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.16.2-arch1-1

Any suggestions?

**Comment by nicoespeon** [comment]
Hey @marc-fez 👋 

Sorry for not replying earlier. I haven't been using the extension in a while, so I'm a bit rusty here. 
I just tried it and "it worked for me" (damn it). 

Now, that error message means that for some reason, the command wasn't registered to VS Code.

Here is where it's declared: https://github.com/nicoespeon/vscode-slides/blob/15d41a0ec3dc06535a552362070fd0aab1cf57cc/package.json#L67-L79

And here is where it's registered: https://github.com/nicoespeon/vscode-slides/blob/15d41a0ec3dc06535a552362070fd0aab1cf57cc/src/extension.ts#L11-L13

I wonder if that could be a race condition or a conflict with another extension that might use the same ID… or something else 😅 

It has been 3 weeks, did you get it to work in the end?

**Comment by dominik-ba** [comment]
I'm having the same issue and don't know how to troubleshoot it

**Comment by nicoespeon** [comment]
Hey @dominik-ba!

Have you just installed the extension? 
Does it make a difference if you restart VS Code?

**Comment by dominik-ba** [comment]
Hi @nicoespeon 

Yes I just installed it and no, a restart does not make any change. Same error message

**Comment by nicoespeon** [comment]
I have made a change to how the extension activates. Now, it will always activate after the editor is ready. That will make your editor startup faster, but should also ensure everything is ready before registering the commands!

I also added an explicit error message if the extension can't register the commands. I noticed it may happen if you open the editor, but there is no workspace folder loaded. If you get this error message, let me know. That means there is a use-case I should handle differently. 

This will be available with v4.2.0 of the extension that should land on VS Code any minute from now!
